package/upgrade/migrations/managed_charts/v1.0.3.sh

```diff
@@ -9,8 +9,62 @@ patch_grafana_resources()
   yq e '.spec.values.grafana.resources = {"limits": {"cpu": "200m", "memory": "500Mi"}, "requests": {"cpu": "100m", "memory": "200Mi"}}' $CHART_MANIFEST -i
 }
 
+patch_alertmanager_enable()
+{
+  # enable alertmanager by default (https://github.com/harvester/harvester-installer/pull/322)
+  yq e '.spec.values.alertmanager.enabled = true' $CHART_MANIFEST -i
+  yq e '.spec.values.alertmanager.config.global.resolve_timeout = "5m"' $CHART_MANIFEST -i
+  yq e '.spec.values.alertmanager.alertmanagerSpec.retention = "120h"' $CHART_MANIFEST -i
+  yq e '.spec.values.alertmanager.alertmanagerSpec.resources = {"limits": {"cpu": "1000m", "memory": "600Mi"}, "requests": {"cpu": "100m", "memory": "100Mi"}}' $CHART_MANIFEST -i
+  yq e '.spec.values.alertmanager.alertmanagerSpec.storage.volumeClaimTemplate.spec.storageClassName = "longhorn"' $CHART_MANIFEST -i
+  yq e '.spec.values.alertmanager.alertmanagerSpec.storage.volumeClaimTemplate.spec.accessModes = ["ReadWriteOnce"]' $CHART_MANIFEST -i
+  yq e '.spec.values.alertmanager.alertmanagerSpec.storage.volumeClaimTemplate.spec.resources.requests.storage = "5Gi"' $CHART_MANIFEST -i
+}
+
+patch_alertmanager_externalurl()
+{
+if [ -n "$HARVESTER_VIP" ]; then
+  # enable alertmanager by default (https://github.com/harvester/harvester-installer/pull/322)
+  PORT=9093
+  yq e '.spec.values.alertmanager.service.port = '$PORT $CHART_MANIFEST -i
+  yq e '.spec.values.alertmanager.alertmanagerSpec.externalUrl = "https://'$HARVESTER_VIP'/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-alertmanager:'$PORT'/proxy/"' $CHART_MANIFEST -i
+fi
+}
+
+patch_prometheus_externalurl()
+{
+if [ -n "$HARVESTER_VIP" ]; then
+  # enable alertmanager by default (https://github.com/harvester/harvester-installer/pull/322)
+  PORT=9090
+  yq e '.spec.values.prometheus.service.port = '$PORT $CHART_MANIFEST -i
+  yq e '.spec.values.prometheus.prometheusSpec.externalUrl = "https://'$HARVESTER_VIP'/api/v1/namespaces/cattle-monitoring-system/services/http:rancher-monitoring-prometheus:'$PORT'/proxy/"' $CHART_MANIFEST -i
+fi
+}
+
+# get harvester vip from configmap, skip potential error
+get_harvester_vip()
+{
+  EXIT_CODE=0
+  #escape the 'return on error'
+  JSON_DATA=$(kubectl get configmap -n kube-system kubevip -o "jsonpath={.data['kubevip-services']}") || EXIT_CODE=$?;
```

Comment:
when the VIP is setup via DHCP the configmap will store an IP address of `0.0.0.0`, may prefer to get it via `kubectl get svc -n kube-system ingress-expose`.

Reply:
good catch, thanks.

updated with:  get VIP from service first, when failure, try configmap